internal/backup/optimizations.go

```
@@ -98,7 +88,10 @@ func deduplicateFiles(ctx context.Context, logger *logging.Logger, root string)
 		}
 
 		rel, relErr := filepath.Rel(root, path)
-		if relErr == nil && shouldSkipDedupPath(rel) {
+		if relErr != nil {
+			return nil
```

golangci-lint (2.12.2)

[error] error is not nil (line 90) but it returns nil

(nilerr)